SSA/Experimental/Bits/Fast/Tactic.lean

```diff
@@ -0,0 +1,301 @@
+import Lean
+import Init.Data.BitVec.Lemmas
+import SSA.Projects.InstCombine.TacticAuto
+import SSA.Projects.InstCombine.LLVM.Semantics
+import SSA.Projects.InstCombine.ForLean
+import Lean.Meta
+import SSA.Experimental.Bits.Fast.FiniteStateMachine
+import SSA.Experimental.Bits.Fast.BitStream
+import SSA.Experimental.Bits.Fast.Decide
+import SSA.Experimental.Bits.Lemmas
+
+open Lean Elab Tactic
+open Lean
+open Lean.Meta
+open Lean.Elab
+open Lean.Elab.Tactic
+
+partial def getFVars  (e : Expr)  : TacticM (List Expr) :=
+    match_expr e with
+      | HAnd.hAnd _ _ _ _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | HSub.hSub _ _ _ _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | HOr.hOr _ _ _ _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | HXor.hXor _ _ _ _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | Eq _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | HAdd.hAdd _ _ _ _ a b => do
+        let a ← getFVars a
+        let b ← getFVars b
+        return a ++ b
+      | Neg.neg _ _ a => do
+        let a ← getFVars a
+        return a
+      | OfNat.ofNat _ _ _ => do
+        return []
+      | _ => match e with
+        | Lean.Expr.fvar a => return [Lean.Expr.fvar a]
+        | _ => throwError s!"getFVars: {e} is not a automata expression"
+
+partial def getVars  (e : Expr)  : TacticM (List Name) :=
+    match_expr e with
+      | HAnd.hAnd _ _ _ _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | HSub.hSub _ _ _ _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | HOr.hOr _ _ _ _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | HXor.hXor _ _ _ _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | Eq _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | HAdd.hAdd _  _ _ _ a b => do
+        let a ← getVars a
+        let b ← getVars b
+        return a ++ b
+      | Neg.neg _ _ a => do
+        let a ← getVars a
+        return a
+      | OfNat.ofNat _ _ _ => do
+        return []
+      | _ => match e with
+        | Lean.Expr.fvar a => return [a.name]
+        | _ => throwError s!"getVars:{e} is not a automata expression"
+
+
+def NatToSyntax (n : Nat) :  (TSyntax `term) :=
```

Comment:
```suggestion
def NatToSyntax (n : Nat) :  Term :=
```
And in general always prefer double backtick'ed names or predefined `TSyntax` abbrevs.